src/main/java/com/flightstats/filesystem/S3FileSystem.java

```diff
@@ -132,7 +145,13 @@ public void flush() throws IOException {
 
         private void doFlush(boolean force) {
             if (initiateMultipartUploadResult == null) {
```

Comment:
Might be nice to pull out this if or the `ObjectMetadata` building into a new method.